skbio/stats/composition.py

```diff
@@ -608,10 +611,329 @@ def centralize(mat):
 
     """
     mat = closure(mat)
-    cen = ss.gmean(mat, axis=0)
+    cen = scipy.stats.gmean(mat, axis=0)
     return perturb_inv(mat, cen)
 
 
+@experimental(as_of="0.4.0-dev")
+def ancom(table, grouping,
+          alpha=0.05,
+          tau=0.02,
+          theta=0.1,
+          multiple_comparisons_correction=None,
+          significance_test=None):
+    r""" Performs a differential abundance test using ANCOM
+
+    This is done by calculating pairwise log ratios between all features
+    and performing a significance test to determine if there is a significant
+    difference in feature ratios with respect to the variable of interest.
+
+    In an experiment with only two treatments, this test tests the following
+    hypothesis for feature :math:`i`
+
+    .. math::
+
+        H_{0i}: \mathbb{E}[\ln(u_i^{(1)})] = \mathbb{E}[\ln(u_i^{(2)})]
+
+    where :math:`u_i^{(1)}` is the mean abundance for feature :math:`i` in the
+    first group and :math:`u_i^{(2)}` is the mean abundance for feature
+    :math:`i` in the second group.
+
+    Parameters
+    ----------
+    table : pd.DataFrame
+       A 2D matrix of counts or proportions where the rows correspond to
+       samples and the columns correspond to features.
+    grouping : pd.Series
+        Vector indicating the assignment of objects to groups.  For example,
+        these could be strings or integers denoting which group an object
+        belongs to.
+    alpha : float, optional
+       Significance level for each of the statistical tests.
+    tau : float, optional
+       A constant used to determine an appropriate cutoff.
+       A value close to zero indicates a conservative cutoff.
+       This can can be anywhere between 0 and 1 exclusive.
+    theta : float, optional
+       Lower bound for the proportion for the W-statistic.
+       If all W-statistics are lower than theta, then no features
+       will be detected to be differentially significant.
+       This can can be anywhere between 0 and 1 exclusive.
+    multiple_comparisons_correction : {None, 'holm-bonferroni'}, optional
+       The multiple comparison correction procedure to run.  If None,
+       then no multiple comparison correction procedure will be run.
+       If 'holm-boniferroni' is specified, then the Holm-Boniferroni
+       procedure [2]_ will be run.
+    significance_test : function, optional
+       A statistical significance function to test for significance between
+       classes.  This function must be able to accept multiple array_like
```

Comment:
As a user I still don't understand what requirements my test function must meet to be used here. What we need to know is the number of arguments and their expected types (described as precisely as possible), and what the function must return (also with precise description).

`array_like` just means that an input can be coerced into an `ndarray` but doesn't state anything about the dimensionality, what `dtype` is expected, etc.